### ISSUE TITLE
Adding enableDetailsRow in CrudController causes column sorting issues

### ISSUE THREAD

**Issue by nlenkowski** [Laravel-Backpack/CRUD#1326]
# Bug report
Hello. I think I just ran into issue #618 despite it being closed. When I add enableDetailsRow() to my CrudController sorting becomes off by one column. Without it enabled sorting works as it should. Is it possible there has been a regression in Backpack\CRUD 3.4? 

Here is my CrudController for reference:
https://pastebin.com/PRMrkN2L

Here is a screencast with enableDetailsRow() enabled and disabled in the CrudController:
https://youtu.be/ifKRFTcmP0s

### What I did:
Enabled enableDetailsRow() on a CrudController.

### What I expected to happen:
Data tables columns sort correctly.

### What happened:
The column directly to the left of the column being clicked is sorted.

### What I've already tried to fix it:
Temporarily disabled enableDetailsRow(), which fixes the sorting issue but disables needed functionality.

### Backpack, Laravel, PHP, DB version:
Backpack\Base 0.9.2
Backpack\CRUD 3.4.4
Laravel 5.6.14
PHP 7.1.15
MySQL 5.7.21

**Comment by indra1** [comment]
This was an issue back when the ajax code for the tables was different. Are you sure your list.blade.php is up to date? Did you make a custom one?

**Comment by nlenkowski** [comment]
Hi. As far as I know I'm using the list.blade.php that shipped with Backpack\CRUD 3.4.4. I have created three custom templates, but list.blade.php is not one of them:

![emailscape 2018-04-20 17-53-06](https://user-images.githubusercontent.com/1290235/39061237-f1508730-44c3-11e8-9ece-daad5aa08907.png)

**Comment by nlenkowski** [comment]
In `/vendor/backpack/crud/src/app/Http/Controllers/CrudFeatures/AjaxTable.php` commenting out the following lines fixes the issue for me:

```
if ($this->crud->details_row) {
       $column_number = $column_number - 1; 
  }
```

Sorting is now working as it should on my CRUD controllers, both with enableDetailsRow() enabled and disabled.

This code above is decrementing the column number by one when a details row is enabled. However, the "+" toggle element that expands the details row is not being added in a new column, it is simply a span element prepended to the first column value. I don't see why the column number should be decremented if a column hasn't actually been added... perhaps this is a bug after all? Any thoughts on this @indra1?

See here:

![image](https://user-images.githubusercontent.com/1290235/39861767-eae34922-5441-11e8-9f3b-28af6c66e226.png)

Big thanks to "Dmitry from Russia" for discovering the fix.

**Comment by tabacitu** [comment]
Hi @nlenkowski ,

Wow - I had no idea this was happening, thanks a lot. Yes, that piece of code seems a leftover from 3.3 details_row functionality. Back then the + sign was inserted as a column, not as span in the first column. So we had to apply this trick. But this is no longer needed, you are right, and it was breaking the sorting.

I've just pushed your fix on ```master```. It'll be available with a ```composer update``` later today, after I tag it with a few more non-breaking changes (it'll be 3.4.10).

Thanks a lot for discovering this, then actually digging down and finding a fix. Much appreciated. Cheers!
